.github/CONTRIBUTING.md

```diff
@@ -361,33 +385,76 @@ This file defines the component properties:
 
 This file defines the input and output schema of each task:
 
-- `title` and `instillShortDescription` will be used by the frontend to provide
-  information about the task.
-- For each property within the `input` and `output` objects:
-  - `instillUIOrder` defines the order in which the properties will be rendered
-    by the frontend.
-  - `required` properties will appear at the forefront of the component UI.
-    Optional properties can be set in the advanced configuration.
-  - `instillUpstreamTypes` define how an input property can be set: the direct
-    value, a reference to another value in the pipeline (e.g.
-    `${trigger.name}` or a combination of both (`my dear ${trigger.name}`).
-
-See the [example recipe](#example-recipe) to see how these fields map to the recipe
-of a pipeline when configured to use this operator.
-
+**Properties within a Task**
+
+- **`title`** is used by Console to provide the title of the task.
+- **`description`** and **`instillShortDescription`** are used by Console to
+  provide a description of the task. If `instillShortDescription` does not
+  exist, it will be the same as `description`.
+- **`input`** is a JSON schema that describes the input of the task.
+- **`output`** is a JSON schema that describes the output of the task.
+
+**Properties within `input` and `output` Objects**
+
+- **`required`** indicates whether the property is required.
+- **`type`**: describes the JSON type of this field, which could be `integer`,
+  `number`, `boolean`, `string`, `array`, or `object`.
+- **`title`** is used by the Console to provide the title of the property.
+- **`description`** is used by Console to show this description in a popup when
+  you click the question mark beside the field.
+- **`instillShortDescription`**: is used by Console to show this under the field
+  (the original description is shown only when hovered). If this value does not
+  exist, it will be the same as the description.
```

Comment:
Maybe this is a bit too specific? I'm thinking that a slight change in the UX might outdate this documentation. Perhaps we can be a bit more generic:

```suggestion
- **`description`** is used by Console to provide information about this task.
- **`instillShortDescription`**: is a concise version of `description`, used to
  fit smaller spaces such as a component form field. If this value is empty,
  the `description` value will be used.
```

What do you think?

Reply:
Also, there's a click vs hover contradiction between both descriptions.

Reply:
Agree, removing the behavior in Console is better.